maevsi/maevsi
# feat(ssr): account for timezone

Closes #999.

This PR implements the following behavior:
- when a request to nuxt is made, the `headers` server middleware tries to get the request origin's time zone by...
    - checking for a cookie
    - querying ip-api.com as a fallback
- the result is written to the requests (!) headers
- a new `dateTime` composable then uses that header to provide a `dayjs` instance being set to the timezone provided

The very first request (that has no cookie set yet) against a production deployment (/ a non-local ip) should therefore guess the timezone using `ip-api.com` for server side rendering. Then, on client side, a cookie is set with a better guess generated locally. All following requests skip fetching `ip-api.com` in favor of the cookie which is available then.

How to do the review test:
1. be on `master` 
1. start maevsi_stack
1. go to an event page and force reload it
1. watch the event's start time display an incorrect value first and watch it quickly change to the correct time (client side hydration)
1. checkout this PR's branch
1. force reload the event page again
1. observe the same behavior as the new cookie has not been available and the request originates from a local source, for which `ip-api.com` provides no data
1. force reload the page again, which now uses the new cookie that was set *after* the page was loaded once using this branche's feature
1. see the correct time being displayed instantly and observe no changes to it

Due to [a recent bug in node](https://github.com/iamkun/dayjs/issues/2102#issuecomment-1324723109) **ensure to have an up-to-date docker image running for development**. To be sure, run a `dargstack build` in this repo's root directory, then redeploy `maevsi_stack`.
